src/kpro_varint.erl

```diff
@@ -0,0 +1,72 @@
+%%%   Copyright (c) 2018, Klarna AB
+%%%
+%%%   Licensed under the Apache License, Version 2.0 (the "License");
+%%%   you may not use this file except in compliance with the License.
+%%%   You may obtain a copy of the License at
+%%%
+%%%       http://www.apache.org/licenses/LICENSE-2.0
+%%%
+%%%   Unless required by applicable law or agreed to in writing, software
+%%%   distributed under the License is distributed on an "AS IS" BASIS,
+%%%   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%%%   See the License for the specific language governing permissions and
+%%%   limitations under the License.
+
+-module(kpro_varint).
+
+-export([ encode/1
+        , decode/1
+        ]).
+
+-define(MAX_BITS, 63).
+
+%% @doc Decode varint.
+-spec decode(binary()) -> {integer(), binary()}.
+decode(Bin) ->
+  dec_zigzag(dec_varint(Bin)).
+
+%% @doc Encode varint.
+-spec encode(kpro:int64()) -> binary().
+encode(Int) ->
+  Bytes = enc_varint(enc_zigzag(Int)),
+  iolist_to_binary(Bytes).
+
+-spec enc_zigzag(integer()) -> integer().
+enc_zigzag(Int) ->
+  true = (Int >= -(1 bsl ?MAX_BITS)),
+  true = (Int < ((1 bsl ?MAX_BITS) - 1)),
+  (Int bsl 1) bxor (Int bsr ?MAX_BITS).
```

Comment:
I assume this one has been tested and found to do the right thing, especially for the extreme values?

Reply:
this is copied from `erlavro` which is tested.